packages/asset-kit-block/src/AssetKitBlock.tsx

```diff
@@ -3,23 +3,63 @@
 import { joinClassNames, useGuidelineDesignTokens } from '@frontify/guideline-blocks-shared';
 import '@frontify/fondue-tokens/styles';
 import { BlockProps } from '@frontify/guideline-blocks-settings';
-import { ReactElement, useState } from 'react';
+import { ReactElement, useEffect, useRef, useState } from 'react';
 import 'tailwindcss/tailwind.css';
-import { useBlockAssets, useBlockSettings, useEditorState } from '@frontify/app-bridge';
+import {
+    BulkDownloadState,
+    useBlockAssets,
+    useBlockSettings,
+    useBulkDownload,
+    useEditorState,
+} from '@frontify/app-bridge';
 import { ASSET_SETTINGS_ID } from './settings';
 import { Settings } from './types';
 import { AssetGrid, AssetSelection, DownloadMessage, InformationSection } from './components';
-import { blockStyle, generateBulkDownload } from './helpers';
+import { blockStyle } from './helpers';
 
 export const AssetKitBlock = ({ appBridge }: BlockProps): ReactElement => {
+    const screenReaderRef = useRef<HTMLDivElement>(null);
     const { designTokens } = useGuidelineDesignTokens();
     const [blockSettings, setBlockSettings] = useBlockSettings<Settings>(appBridge);
     const isEditing = useEditorState(appBridge);
     const { blockAssets, addAssetIdsToKey, deleteAssetIdsFromKey, updateAssetIdsFromKey } = useBlockAssets(appBridge);
     const [isUploadingAssets, setIsUploadingAssets] = useState<boolean>(false);
-    const [isDownloadingAssets, setIsDownloadingAssets] = useState<boolean>(false);
-    const { title, description, hasBorder_blocks, hasBackgroundBlocks } = blockSettings;
+    const { title, description, hasBorder_blocks, hasBackgroundBlocks, downloadUrlBlock } = blockSettings;
     const currentAssets = blockAssets[ASSET_SETTINGS_ID] ?? [];
+    const { generateBulkDownload, status, downloadUrl } = useBulkDownload(appBridge);
+
+    const startDownload = () => {
+        if (downloadUrlBlock) {
+            downloadAssets(downloadUrlBlock);
+        } else {
+            generateBulkDownload(currentAssets.map((asset) => asset.id));
+        }
+    };
+
+    const downloadAssets = (downloadUrl: string) => {
+        window.open(downloadUrl, '_blank');
```

Comment:
Does this open a new tab that closes automatically? If yes, can we go with an approach like in `downloadAsset.ts` ?

Reply:
Yes it does, and I adjusted it 👍